### PR TITLE
ExUnit.Runner - Pass config seed and not the whole config when not need

### PR DESCRIPTION
**Pull request by eksperimental**
Avoid passing the whole config when only the seed is needed

### CODE REVIEW

File: lib/ex_unit/lib/ex_unit/runner.ex [hunk 1]
```diff
@@ -285,15 +285,15 @@ defmodule ExUnit.Runner do
   defp spawn_test(config, test, context) do
     parent_pid = self()
     timeout = get_timeout(config, test.tags)
-    {test_pid, test_ref} = spawn_test_monitor(config, test, parent_pid, context)
+    {test_pid, test_ref} = spawn_test_monitor(test, parent_pid, context, config.seed)
```
Comment (fertapric) [inline lib/ex_unit/lib/ex_unit/runner.ex]: I'm 👎 for this change. I liked better when `spawns_test/3` did not know that the only thing that `spawn_test_monitor/4` needs is the seed configuration.

**Comment by josevalim** [comment]
Agreed with @fertapric :) Thanks @eksperimental !

File: lib/ex_unit/lib/ex_unit/runner.ex (hunk 1 above)
Comment (eksperimental) [reply lib/ex_unit/lib/ex_unit/runner.ex]: yeah, it's understandable.
the original idea was the replace generate_test_seed, which only needs the seed.

**Comment by eksperimental** [comment]
@josevalim could you please reopen the PR? I will submit a new one just changing `generate_test_seed`

**Comment by josevalim** [comment]
I would say it is also fine as is. :) thanks!
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Pull request closed without merging**